### PR TITLE
[Feature] Country Code Picker

### PR DESCRIPTION
**Pull request by shubhamsinghshubham777**
Click on the image to see the preview demo gif:

Country Picker preview

### CODE REVIEW

File: app/src/main/java/dev/baseio/discordjetpackcompose/ui/components/CountryPicker.kt [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+import dev.baseio.discordjetpackcompose.utils.readAssetFile
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun CountryPicker(
+    bottomSheetScaffoldState: ModalBottomSheetState,
+    backgroundContent: @Composable () -> Unit,
+    onCountrySelected: (Country) -> Unit,
+    countryList: List<Country>? = LocalContext.current.readAssetFile(Constants.CountriesAssetFilePath)
```
Comment (Anmol92verma) [inline app/src/main/java/dev/baseio/discordjetpackcompose/ui/components/CountryPicker.kt:67]: @shubhamsinghshubham777 this is an IO op , lets move these to VM

File: app/src/main/java/dev/baseio/discordjetpackcompose/ui/components/CountryPicker.kt [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                    )
+                    val lazyListState = rememberLazyListState()
+                    LazyColumn(
+                        state = lazyListState,
+                        modifier = Modifier.simpleVerticalScrollbar(
+                            state = lazyListState,
+                            color = MaterialTheme.colors.onSurface.copy(alpha = 0.5f)
+                        ),
+                    ) {
+                        countryList?.filter { country ->
```
Comment (Anmol92verma) [inline app/src/main/java/dev/baseio/discordjetpackcompose/ui/components/CountryPicker.kt:90]: @shubhamsinghshubham777 move this to VM

File: app/src/main/java/dev/baseio/discordjetpackcompose/ui/components/CountryPicker.kt [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+@Composable
+private fun CountryItem(
+    countryName: String,
+    countryCode: String,
+    onCountrySelected: () -> Unit,
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .clickable(
```
Comment (Anmol92verma) [inline app/src/main/java/dev/baseio/discordjetpackcompose/ui/components/CountryPicker.kt:121]: @shubhamsinghshubham777 can we have a global extension of this modifier which is called clickableWithRipple ?

File: app/src/main/java/dev/baseio/discordjetpackcompose/ui/components/CountryPicker.kt [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        horizontalArrangement = Arrangement.SpaceBetween,
+    ) {
+        Text(text = countryName)
+        Text(text = countryCode, modifier = Modifier.alpha(ContentAlpha.medium))
+    }
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+private fun SearchBox(currentValue: String, onValueChanged: (String) -> Unit) {
```
Comment (Anmol92verma) [inline app/src/main/java/dev/baseio/discordjetpackcompose/ui/components/CountryPicker.kt:135]: @shubhamsinghshubham777 good to have this moved to a separate file

File: app/src/main/java/dev/baseio/discordjetpackcompose/ui/components/CountryPicker.kt [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+    DiscordJetpackComposeTheme {
+        val sheetState = rememberModalBottomSheetState(initialValue = ModalBottomSheetValue.Hidden)
+        val coroutineScope = rememberCoroutineScope()
+        CountryPicker(bottomSheetScaffoldState = sheetState, backgroundContent = {
+            Box(modifier = Modifier
+                .fillMaxSize()
+                .clickable {
+                    coroutineScope.launch { sheetState.show() }
+                }) {
+                Text(text = "Click anywhere to open the country picker...")
```
Comment (Anmol92verma) [inline app/src/main/java/dev/baseio/discordjetpackcompose/ui/components/CountryPicker.kt:185]: @shubhamsinghshubham777 can we use stringResource(R.string. ???

File: app/src/main/java/dev/baseio/discordjetpackcompose/ui/utils/ComposeUtils.kt [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    )
+
+    drawWithContent {
+        drawContent()
+
+        val firstVisibleElementIndex = state.layoutInfo.visibleItemsInfo.firstOrNull()?.index
+        val needDrawScrollbar = state.isScrollInProgress || alpha > 0.0f
+
+        // Draw scrollbar if scrolling or if the animation is still running and lazy column has content
+        if (needDrawScrollbar && firstVisibleElementIndex != null) {
```
Comment (Anmol92verma) [inline app/src/main/java/dev/baseio/discordjetpackcompose/ui/utils/ComposeUtils.kt:36]: @shubhamsinghshubham777 looks good, we did not have a scrollbar in lazyColumn ? 🤔 ❤️

File: data/src/main/java/dev/baseio/discordjetpackcompose/models/Country.kt [hunk 7]
```diff
@@ -0,0 +1,19 @@
+package dev.baseio.discordjetpackcompose.models
+
+
+import androidx.annotation.Keep
+import com.google.gson.annotations.SerializedName
+
+@Keep
+data class Country(
```
Comment (Anmol92verma) [inline data/src/main/java/dev/baseio/discordjetpackcompose/models/Country.kt:8]: @shubhamsinghshubham777 also let's move the logic to fetch countries in data module

File: data/src/main/java/dev/baseio/discordjetpackcompose/utils/Constants.kt [hunk 8]
```diff
@@ -0,0 +1,5 @@
+package dev.baseio.discordjetpackcompose.utils
+
+object Constants {
```
Comment (Anmol92verma) [inline data/src/main/java/dev/baseio/discordjetpackcompose/utils/Constants.kt:3]: @shubhamsinghshubham777 move this constant to the repository impl where we will have the logic to fetch the countries.

File: data/src/main/java/dev/baseio/discordjetpackcompose/utils/ContextUtils.kt [hunk 9]
```diff
@@ -0,0 +1,17 @@
+package dev.baseio.discordjetpackcompose.utils
+
+import android.content.Context
+import android.util.Log
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+
+inline fun <reified T> Context.readAssetFile(filePath: String): T? {
```
Comment (Anmol92verma) [inline data/src/main/java/dev/baseio/discordjetpackcompose/utils/ContextUtils.kt:8]: @shubhamsinghshubham777 can we call this file ExtensionUtils ?

File: data/src/main/java/dev/baseio/discordjetpackcompose/utils/ContextUtils.kt [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+    return try {
+        val jsonDataFromFile = this.assets.open(filePath).bufferedReader().use { it.readText() }
+        val returnType = object : TypeToken<T>() {}.type
+        Gson().fromJson(jsonDataFromFile, returnType)
+    } catch (e: Exception) {
+        Log.e("ContextUtils", "Some error occurred. Reason:", e)
```
Comment (Anmol92verma) [inline data/src/main/java/dev/baseio/discordjetpackcompose/utils/ContextUtils.kt:14]: @shubhamsinghshubham777 let's log this using Timber

File: app/src/main/java/dev/baseio/discordjetpackcompose/ui/utils/ComposeUtils.kt (hunk 6 above)
Comment (shubhamsinghshubham777) [reply app/src/main/java/dev/baseio/discordjetpackcompose/ui/utils/ComposeUtils.kt:40]: Thanks. Yes, unfortunately we didn't have it by default.

File: data/src/main/java/dev/baseio/discordjetpackcompose/models/Country.kt (hunk 7 above)
Comment (shubhamsinghshubham777) [reply data/src/main/java/dev/baseio/discordjetpackcompose/models/Country.kt:8]: Do you mean we need to create a usecase for fetching countries?

**Pull request merged**